p2p/events.py

```diff
@@ -1,12 +1,24 @@
 from typing import (
     Type,
+    TYPE_CHECKING,
 )
 
 from lahja import (
     BaseEvent,
     BaseRequestResponseEvent,
 )
 
+if TYPE_CHECKING:
+    from p2p.peer import BasePeer
+
+
+class PeerConnectedEvent(BaseEvent):
+    """
+    Broadcasted when a new peer of any kind connects to the peer pool.
+    """
+    def __init__(self, peer: 'BasePeer'):
```

Comment:
I would expect that we can't transmit peer instances over multiprocessing pipes...  am I missing something?

Reply:
Every event that travels across processes (that is every event except those that are `BroadcastConfig(internal=True)`) needs to be pickable. I assume that a `BasePeer` would not be pickable.

My vision for how these things should work is basically that we would create shallow DTOs for this purpose which can be matched back to the original peer based on some unique identifier ([see also](https://github.com/ethereum/py-evm/issues/1198#issuecomment-417740431)).